### ISSUE TITLE
wubi

### ISSUE THREAD

**Issue by farhan-zak** [hakuna-m/wubiuefi#32]
I want to install ubuntu 16.04 lts alongside with windows 8.1 which one of the wubi installers should i use ?
please help.

**Comment by hakuna-m** [comment]
A typical installation "alongside Windows" is an installation without Wubi. Because the installer Wubi shares a Windows drive/partition with Windows by default. Official name of a typical Wubi installation is "Ubuntu within Windows".

If "alongside" means only for you that you don't want to overwrite Windows, you can use Wubiuefi, of course.

Currently, for Ubuntu 16.04 LTS I would use the first point release 16.04.1. If you didn't download a matching ISO (e.g. [ubuntu-16.04.1-desktop-amd64.iso](http://releases.ubuntu.com/16.04.1/ubuntu-16.04.1-desktop-amd64.iso) for 64 bit) and you don't need features of the latest release, I recommend latest Wubiuefi version for 16.04.1 from [release table](https://github.com/hakuna-m/wubiuefi/wiki#releases). Currently, [16041r311](https://github.com/hakuna-m/wubiuefi/releases/tag/16041r311).

If you downloaded an ISO and you need features of the latest release, it is also possible to use (latest release)[https://github.com/hakuna-m/wubiuefi/releases/latest]. Note: Automatic download doesn't work and Ubuntu 16.04.1 64 bit is desktop environment "Wubi EFI" and not "Ubuntu" in this case. "Ubuntu" would be the desktop environment for the matching Ubuntu version. e.g. for 1610r319 it is Ubuntu 16.10.

**Comment by farhan-zak** [comment]
I wanted to ask whether I can use wubuefi 16.10(rev 319) for ubuntu 16.04LTS or should I use the other one.
What would you suggest ?

**Comment by hakuna-m** [comment]
Currently, I suggest [16041r311](https://github.com/hakuna-m/wubiuefi/releases/tag/16041r311) for 16.04. If there are issues, you can copy the downloaded iso (C:\ubuntu\install\installation.iso) in the same folder as wubi1610r319.exe and try it with this version. But there are only few known issues which r319 can fix in comparison with r311 for 16.04.

**Comment by farhan-zak** [comment]
Should I use other drives to install ubuntu or should I use the basic C drive ?

**Comment by hakuna-m** [comment]
Wubi installation method on a Windows partition has a less robust filesystem (see [Wubi info](http://askubuntu.com/tags/wubi/info), so therefore it is better to use a different drive. But if no other drive exists, it is more risky to create a new drive. So it is up to you. Both options are possible.

**Comment by farhan-zak** [comment]
Which one is better Sir ?

**Comment by hakuna-m** [comment]
Does another Windows drive exist on your system?

**Comment by farhan-zak** [comment]
Yes it exists. Should I make separate partition of 100GB for ubuntu ?

**Comment by hakuna-m** [comment]
A separate partition is the best option to avoid problems because of the less robust filesystem. But be careful, sometimes Windows adds additional partitions by converting to [Dynamic Disks](https://msdn.microsoft.com/en-us/library/windows/desktop/aa363785(v=vs.85).aspx). Dynamic Disks don't work with Wubi. Additionally, note that Wubiuefi doesn't give you the option to use 100GB for system files with default settings.  Either it is necessary to increase disk image for system files (root.disk) or to use the additional space for non system files.

**Comment by hakuna-m** [comment]
Some additional notes regarding "less robust file system".
 
For testing purpose I installed a Wubi installation on a notebook with a pre-installed Windows 8.1 in 2013.  I selected Windows system drive C: as location because I wanted to test an installation without changing partitions. Since then, I have used it frequently to test stability of Wubi installations. 
I upgraded it to 14.04 and I increased space of root.disk if necessary. But there were no issues because of a "less robust file system" in 2013, 2014 and 2015.

In the middle of 2016, I needed the notebook for a video software which is only available for Windows. 
I connected an external USB hard drive and created and also removed large files on external drive on Windows. The large files were not on C: drive but IMHO some temporary files and system files are always on C: drive under Windows by default. 

One time Windows said after booting that an error occured and it was necessary to check and repair file system. After repairing C:\ubuntu\disks\root.disk had zero bytes and of course, Ubuntu didn't work.

Unfortunately, I had not enough time to debug the issue. I restored a previously saved copy of root.disk. Since then, the installation has worked without file system issues.

So IMHO it is better that you don't use C: drive to avoid this specific issue of Windows file repairing. But in general, issues are rare.

**Comment by farhan-zak** [comment]
Thanks Sir.
